xbmc/cores/VideoPlayer/Buffers/VideoBufferDMA.cpp

```diff
@@ -119,20 +119,29 @@ bool CVideoBufferDMA::Alloc()
 
 void CVideoBufferDMA::Export(AVFrame* frame, uint32_t width, uint32_t height)
 {
-  m_planes = av_pix_fmt_count_planes(static_cast<AVPixelFormat>(frame->format));
+  AVPixelFormat pix_fmt = static_cast<AVPixelFormat>(frame->format);
+  m_planes = av_pix_fmt_count_planes(pix_fmt);
+  int h_shift, v_shift;
+
+  if (av_pix_fmt_get_chroma_sub_sample(pix_fmt, &h_shift, &v_shift))
+    throw std::runtime_error(
+        "unable to determine chroma_sub_sample: " +
+        std::string(av_get_pix_fmt_name(pix_fmt)));
 
   if (m_planes < 2)
     throw std::runtime_error(
         "non-planar formats not supported: " +
-        std::string(av_get_pix_fmt_name(static_cast<AVPixelFormat>(frame->format))));
+        std::string(av_get_pix_fmt_name(pix_fmt)));
 
   for (uint32_t plane = 0; plane < m_planes; plane++)
-  {
     m_strides[plane] =
         av_image_get_linesize(static_cast<AVPixelFormat>(frame->format), width, plane);
+
+  m_offsets[0] = 0;
+  m_offsets[1] = m_strides[0] * height;
+  for (uint32_t plane = 2; plane < m_planes; plane++)
     m_offsets[plane] =
```

Comment:
Off by one error I think?

Reply:
Could you explain exactly what you think the correction is? It's looking okay to me.
plane(N) = offset_of_plane(N-1) + size_of_plane(N-1).
And we've already done plane(0) and plane(1).

Reply:
Yea I ran some iterations and it seems fine. I don't think we need the loop at all though and we can just be explicit about what is happening
```c++
  m_offsets[0] = 0;
  m_offsets[1] = m_offsets[0] + (m_strides[0] * (height >> 0));
  m_offsets[2] = m_offsets[1] + (m_strides[1] * (height >> v_shift));
```

There may not always be three planes but we allocate 3 always.

Reply:
In theory there could be formats with an additional plane (e.g. alpha), so I left the loop in, but as you say it is initialised to 3, and handling a format with extra planes may well require extra logic so I've removed it.

I've made sanity check of `m_planes >= 2` also require `<=3`.